ts/private/npm_repositories.bzl

```diff
@@ -45,6 +45,10 @@ def _http_archive_version_impl(rctx):
     rctx.download_and_extract(
         url = [u.format(version) for u in rctx.attr.urls],
         integrity = integrity,
+        # Prevents accidental re-use of cached versions that would otherwise
+        # be used purely based on the "integrity" value. E.g. someone forgot
+        # to update the integrity but the `ts_version` is already different.
+        canonical_id = "%s_%s" % (version, integrity),
```

Comment:
Should we use https://bazel.build/rules/lib/repo/cache#get_default_canonical_id instead of inventing our own scheme?

Reply:
Should we at least use the same implementation as [get_default_canonical_id](https://bazel.build/rules/lib/repo/cache#get_default_canonical_id)? Also put a TODO to use it once we drop bazel6...

Reply:
> You might also ask, why does Bazel not call this default helper by default.

That's a really good question 😢

Reply:
But I'm just looking at `get_default_canonical_id` as an example of what `canonical_id` should normally be used for.

I think we basically want to invalidate the cache if the URL changes, even if the `integrity` did not change? That's also what `get_default_canonical_id` does... 🤷

Reply:
I'm good using that, so I'll try switch to it;